test/inductor/test_torchinductor.py

```diff
@@ -466,14 +468,17 @@ def reference_to_expect(actual_flat, correct_flat):
         correct = tree_unflatten(correct_flat, correct_spec)
 
     if assert_equal:
-        self.assertEqual(
-            actual,
+        comparator_assert(
+            self,
+            ref_inputs,
+            example_inputs,
             correct,
+            actual,
             atol=atol,
             rtol=rtol,
-            equal_nan=True,
             exact_dtype=exact_dtype,
         )
+
         # In case of input mutations, check that inputs are the same
         self.assertEqual(
```

Comment:
Should you move this into the default checker too?

Reply:
I think the input mutation check should be ran regardless of which checker is being used, so to avoid duplicating it I moved it outside of the checker functions to avoid duplicating it. Would you prefer it be moved into the comparison functions to simplify that part?

Reply:
It also applied to `sort_equal_checker`? I saw you have special logic for input checks there?